### PR TITLE
Input liseners truncated with getLat() and getLon()

### LINKED ISSUE #61: ensure displayed coordinates are truncated to precision
under `Option 3 : By giving co-ordinates` -- thanks!

### PR DESCRIPTION
**Pull request by mridulnagpal**
Close #61 and related to start of #63

### CODE REVIEW

File: src/blurredLocation.js [hunk 1]
```diff
@@ -214,24 +210,42 @@ BlurredLocation = function BlurredLocation(options) {
     options.map.off('moveend',updateRectangleOnPan);
   }
 
-  var marker = L.marker([getFullLat(), getFullLon()]);
+  enableCenterShade();
 
-  function updateMarker() {
-    if(marker) marker.remove();
-    marker = L.marker([getFullLat(), getFullLon()]).addTo(options.map);
+  function enableLatLngInputTruncate() {
```
Comment (jywarren) [inline src/blurredLocation.js]: Does this belong in the interface module?

File: src/blurredLocation.js [hunk 2]
```diff
@@ -214,24 +210,42 @@ BlurredLocation = function BlurredLocation(options) {
...
+      }
+    });
   }
 
-  enableCenterShade();
+  function getCountry(result) {
+    if(result) {
+      for (i in result.results) {
+        if(result.results[i].types.indexOf("country") != -1) {
+          console.log(result.results[i].formatted_address);
```
Comment (jywarren) [inline src/blurredLocation.js]: you'll need to remove this i guess?

Comment (jywarren) [inline src/blurredLocation.js]: just the console.log bit

File: src/blurredLocation.js [hunk 3]
```diff
@@ -214,24 +210,42 @@ BlurredLocation = function BlurredLocation(options) {
...
+      $.ajax({
+      url:"https://maps.googleapis.com/maps/api/geocode/json?latlng="+lat+","+lng ,
+      success: function(result) {
+        onResponse(result);
+      }
+    });
   }
 
-  enableCenterShade();
+  function getCountry(result) {
```
Comment (jywarren) [inline src/blurredLocation.js]: Cool - are you planning to add getCity, getProvince/getState, etc?

File: src/ui/Interface.js [hunk 4]
```diff
@@ -27,8 +27,21 @@ module.exports = function Interface (options) {
       if($("#"+options.selector).val())
         $("#location").val($("#"+options.selector).val());
 
-      else if(result.results[0])
-        $("#location").val(result.results[0].formatted_address);
+      else if(result.results[0]) {
+        if(options.getPrecision() == 0) {
+          for (i in result.results) {
+            if(result.results[i].types.indexOf("country") != -1) {
```
Comment (jywarren) [inline src/ui/Interface.js]: Can you add a comment describing what's going on at this part of the code? I find it hard to follow. Thanks!

**Comment by jywarren** [comment]
Once this is ready can you push up a gh-pages demo to try? And bump the version number in `package.json` by a minor version number (see http://semver.org)? Thanks!

File: src/blurredLocation.js (hunk 1 above)
Comment (mridulnagpal) [inline src/blurredLocation.js]: Sure will put this in the interface module.

File: src/blurredLocation.js (hunk 3 above)
Comment (mridulnagpal) [inline src/blurredLocation.js]: Yup

File: src/blurredLocation.js (hunk 2 above)
Comment (mridulnagpal) [inline src/blurredLocation.js]: Sorry was there for testing purposes, removed it!

File: src/ui/Interface.js (hunk 4 above)
Comment (mridulnagpal) [inline src/ui/Interface.js]: Sure

File: dist/Leaflet.BlurredLocation.js [hunk 5]
```diff
@@ -14369,9 +14360,7 @@ BlurredLocation = function BlurredLocation(options) {
   return {
     getLat: getLat,
     getLon: getLon,
-    goTo: goTo,
-    geocodeStringAndPan: geocodeStringAndPan,
-    getSize: getSize,
+    goTo: goTo,    getSize: getSize,
```
Comment (jywarren) [inline dist/Leaflet.BlurredLocation.js]: Separate lines?

**Comment by mridulnagpal** [comment]
gh-pages up and running

File: dist/Leaflet.BlurredLocation.js (hunk 5 above)
Comment (mridulnagpal) [inline dist/Leaflet.BlurredLocation.js]: Done!

**Comment by mridulnagpal** [comment]
The new version will be 1.1.0 right?

**Comment by jywarren** [comment]
Maybe plus some docs in the README about this new stuff?

**Comment by mridulnagpal** [comment]
Added some docs

File: src/ui/Interface.js [hunk 6]
```diff
@@ -27,8 +27,23 @@ module.exports = function Interface (options) {
...
+          //Iterates through all locations available, and checks if the type of location is country
+          for (i in result.results) {
+            if(result.results[i].types.indexOf("country") != -1) {
+              //If the type of location is a country assign it to thr input box value
+              $("#location").val(result.results[i].formatted_address);
+            }
+          }
+        }
+        else {
+          $("#location").val(result.results[0].formatted_address);
```
Comment (jywarren) [inline src/ui/Interface.js]: Hmm, we should make `#location` a parameter in the constructor no? So people can set it?

File: src/blurredLocation.js [hunk 7]
```diff
@@ -259,8 +263,8 @@ BlurredLocation = function BlurredLocation(options) {
     setZoomByPrecision: setZoomByPrecision,
     disableCenterShade: disableCenterShade,
     enableCenterShade: enableCenterShade,
-    disableCenterMarker: disableCenterMarker,
-    enableCenterMarker: enableCenterMarker,
+    test: test,
```
Comment (jywarren) [inline src/blurredLocation.js]: Why is this called test? Sorry, I don't follow.

File: src/ui/Interface.js (hunk 6 above)
Comment (mridulnagpal) [inline src/ui/Interface.js]: Sure, on it!

File: src/blurredLocation.js (hunk 7 above)
Comment (mridulnagpal) [inline src/blurredLocation.js]: Just for a test will tidy up code, right now!!

Comment (mridulnagpal) [inline src/blurredLocation.js]: tidied up the code, and added an option for #location, please have a look!!

File: src/ui/Interface.js [hunk 8]
```diff
@@ -2,7 +2,8 @@ module.exports = function Interface (options) {
 
     options.latId = options.latId || 'lat';
     options.lngId = options.lngId || 'lng';
-    options.selector = options.selector || 'geo_location'
+    options.selector = options.selector || 'geo_location';
+    options.locationText = options.locationText || 'location';
```
Comment (jywarren) [inline src/ui/Interface.js]: This is great. Could we say `placenameId`?

Also, I think there's ambiguity between `geo_location`/`options.selector` and `location`/`locationText` -- can we standardize these to be more understandable? What about:

```js
options.placenameInputId = options.placenameInputId || 'placenameInput'; // the placename as input by the user
options.placenameDisplayId = options.placenameDisplayId || 'placenameDisplay'; // the placename as will be stored/displayed
```

File: src/ui/Interface.js [hunk 9]
```diff
@@ -25,10 +26,25 @@ module.exports = function Interface (options) {
     function changeVal(result) {
```
Comment (jywarren) [inline src/ui/Interface.js]: Could we change this to `onPlacenameReturned` to clarify a bit?

File: src/ui/Interface.js [hunk 10]
```diff
@@ -25,10 +26,25 @@ module.exports = function Interface (options) {
...
 
       if($("#"+options.selector).val())
-        $("#location").val($("#"+options.selector).val());
+        $("#"+options.locationText).val($("#"+options.selector).val());
 
-      else if(result.results[0])
-        $("#location").val(result.results[0].formatted_address);
+      else if(result.results[0]) {
+        if(options.getPrecision() <=0 ) {
```
Comment (jywarren) [inline src/ui/Interface.js]: Hmm, here, what if we simply made the request to `getPlacenameFromCoordinates()` for a specific "resolution" -- like, 

```js
getPlacenameFromCoordinates(options.getLat(), options.getLon(), options.getPrecision(), onPlacenameReturned)
```

and we deal with the different types in the `getPlacenameFromCoordinates()` definition? That way, if people wanted to connect up a different geocoder, and the response were differently formatted, we wouldn't have to worry about it in the interface code over here. `getPlacenameFromCoordinates()` could return a string, no more.

File: src/ui/Interface.js (hunk 8 above)
Comment (mridulnagpal) [inline src/ui/Interface.js]: Done

File: src/ui/Interface.js (hunk 9 above)
Comment (mridulnagpal) [inline src/ui/Interface.js]: Done

File: src/ui/Interface.js (hunk 10 above)
Comment (mridulnagpal) [inline src/ui/Interface.js]: Done, beginning with state part

**Comment by mridulnagpal** [comment]
@jywarren I made a function of my own to sort through the result of the API, basically what `result_type` did for the API, I made myself.

**Comment by mridulnagpal** [comment]
I have reworked all the things you told me and also now the location display works properly, displaying only the location depth which is required. Pushing everything to gh-pages

**Comment by jywarren** [comment]
Can you resolve the conflicts and confirm it's still working? Then we can merge in. Thanks!

**Comment by mridulnagpal** [comment]
No more conflicts :smile:

**Comment by jywarren** [comment]
Did u rebuild it? Thanks!

File: src/blurredLocation.js [hunk 11]
```diff
@@ -97,11 +98,28 @@ BlurredLocation = function BlurredLocation(options) {
...
+                //If the type of location is a country assign it to thr input box value
+                country = result.results[i].formatted_address;
+              }
+            }
+
+            if(precision <= 0) onResponse(country);
+
+            else if(precision == 1) onResponse(fullAddress[fullAddress.length - 2] + "," + country);
+
+            else if(precision >= 2) onResponse(fullAddress[fullAddress.length - 3] + "," + fullAddress[fullAddress.length - 2] + "," + country);
```
Comment (jywarren) [inline src/blurredLocation.js]: We should also ensure that if precision is not set, there's a default -- probably just the full address.

Comment (mridulnagpal) [inline src/blurredLocation.js]: Done!! Ready to merge?

**Comment by jywarren** [comment]
Yep!!!
